### PR TITLE
[Customer Center] Build feedback survey from JSON

### PR DESCRIPTION
**Pull request by vegaro**
Based off #3933 

https://github.com/RevenueCat/purchases-ios/assets/664544/11eae984-294a-4e14-8c40-7c2d50994c09

Can probably use some animations, but tuning that up will come up later

It will open a feedback survey for an option if there's one

### CODE REVIEW

**Comment by vegaro** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/RevenueCat/purchases-ios/3959?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3959** <a href="https://app.graphite.dev/github/pr/RevenueCat/purchases-ios/3959?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3933** <a href="https://app.graphite.dev/github/pr/RevenueCat/purchases-ios/3933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3919** <a href="https://app.graphite.dev/github/pr/RevenueCat/purchases-ios/3919?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3966** <a href="https://app.graphite.dev/github/pr/RevenueCat/purchases-ios/3966?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vegaro and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@available(visionOS, unavailable)
+struct FeedbackSurveyButtonsView: View {
+
+    let options: [CustomerCenterConfigData.HelpPath.FeedbackSurvey.Option]
+    let action: (() -> Void)
+
+    var body: some View {
+        VStack(spacing: 16) {
```
Comment (tonidero) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:57]: Should we extract this to some constant?

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            ForEach(options, id: \.id) { option in
+                Button(option.title) {
+                    Task {
+                        self.action()
```
Comment (tonidero) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:61]: Hmm so the way it is right now, every option will have the same action. However, we might have different actions right? Like some questions might provide a promotional offer and some won't. Do we need to change this and/or send back the id?

Comment (tonidero) [inline RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:61]: A different question, if you press the button multiple times quickly, do we handle it gracefully? Mostly thinking about the `Task` we use here so not sure...

**Review by tonidero — APPROVED** [review]
Just some questions but looking good!

File: RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift (hunk 2 above)
Comment (vegaro) [reply RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:61]: You're correct. I built that in https://github.com/RevenueCat/purchases-ios/pull/3968/files#diff-b11601f94f84f34e8c7259de27e1760a2919c2ccb27d582eb7d4daa019d57f37 to keep this one simple, I should have make that more clear sorry!

Comment (vegaro) [reply RevenueCatUI/CustomerCenter/Views/FeedbackSurveyView.swift:61]: Also handled in https://github.com/RevenueCat/purchases-ios/pull/3968/files#diff-b11601f94f84f34e8c7259de27e1760a2919c2ccb27d582eb7d4daa019d57f37R82 although it should be an AsyncButton actually, so I will update that in this PR

**Pull request merged**